### PR TITLE
Ootb webhooks support

### PR DESCRIPTION
**Pull request by chpro**
An implementation to support the webhooks which come with bitbucket server ootb. No plugin needs to be installed.

See atlassian documentation for more information about webhooks and their payload.

https://confluence.atlassian.com/bitbucketserver0510/managing-webhooks-in-bitbucket-server-951390737.html
https://confluence.atlassian.com/bitbucketserver0510/event-payload-951390742.html

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/BitbucketPayloadProcessor.java [hunk 1]
```diff
@@ -27,18 +27,53 @@ public void processPayload(JSONObject payload, HttpServletRequest request) {
...
+    	} else if (payload.has("pullRequest")) { // for all PR events
+    		repo = payload.getJSONObject("pullRequest").getJSONObject("toRef").getJSONObject("repository");
+    	} else {
+    		LOGGER.log(Level.WARNING, "Not possible to trigger job for event '{0}'. Only PR events and pushes are supported for now.", payload.get("eventKey"));
+    		LOGGER.log(Level.FINE, payload.toString());
+    		return;
+    	}
+    	
+        String user = payload.getJSONObject("actor").getString("name");
+        String url = repo.getJSONObject("project").getString("key").toLowerCase() + "/" + repo.getString("name");
```
Comment (dhufnagel) [inline src/main/java/com/cloudbees/jenkins/plugins/BitbucketPayloadProcessor.java]: This does not work on my site. I needed to replace `repo.getString("name")` with `repo.getString("slug")` to make it work. The name part is not equal to the slug part, if there is a whitespace in the name.

Comment (chpro) [reply src/main/java/com/cloudbees/jenkins/plugins/BitbucketPayloadProcessor.java]: thank you will incorporate the changes

**Comment by HerrmannHinz** [comment]
any update on this?

**Comment by krotte1** [comment]
When is this pull-request going to be merged and a new build created. The fixes on this pull-request are desperately needed for anyone using an on-prem version of Bitbucket.

**Comment by krotte1** [comment]
How do we get jenkinsci to pay attention to this plugin. There are valid pull requests out there that need to be merged, but nobody is doing it!! This is extremely frustrating.

**Comment by thismatters** [comment]
@krotte1 you're right, this is frustrating. I would love to see this change incorporated since I have no control over what plugins are installed on our Bitbucket Server.

**Comment by krotte1** [comment]
@fbelzunc I noticed that you have been maintaining this repo through other recent pull request merges... I was wondering if you could review this pull request and get it merged... or give me the permission to do so?

**Comment by krotte1** [comment]
@fbelzunc I and the rest of the on-prem bitbucket/Jenkins users would really love to have this fix merged in. Because of the differences in payload sent by server vs cloud, this plugin is essentially useless without this fix.

**Comment by krotte1** [comment]
@fbelzunc @arcadien @nremond @ndeloof I was really hoping that we could get this pull request merged. Can one of you please do that? It expands support to botbucket server on prem users.

**Comment by fbelzunc** [comment]
I will check this next week.

**Comment by krotte1** [comment]
Thank you @fbelzunc for getting this merged!!!!

**Pull request merged**